### ISSUE TITLE
RedundantParens: Format breaks compile for PAT optionals

### ISSUE THREAD

**Issue by CrownedPhoenix** [nicklockwood/SwiftFormat#1156]
Swift recently introduced the `any` keyword enabling Protocols-as-Types.
The following code snippet, will compile:
```
protocol Thing {}

struct Barbarian {
    let weapon: (any Thing)?
}
```
But when run with `.swiftformat --rules redundantParents .` formats to the following (which does not compile):
```
protocol Thing {}

struct Barbarian {
    let weapon: any Thing?
}
```

**Comment by nicklockwood** [comment]
Sorry, I should've been more on-the-ball with this one. I'll get a fix out ASAP.

**Comment by CrownedPhoenix** [comment]
Honestly, I feel like you get a lot of points just for responding to this in like an hour.
Thanks!

**Comment by nicklockwood** [comment]
@CrownedPhoenix fixed in 0.49.6
